app/main.py

```diff
@@ -1,34 +1,210 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple,
+                 end: tuple, is_drowned: bool = False) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if start[0] == start[1] and end[0] == end[1]:
+            self.start_column, self.end_column = start[1], end[1]
+            self.end_row, self.start_row = end[0], start[0]
```

Comment:
The assignment of `self.start_column, self.end_column` and `self.end_row, self.start_row` seems incorrect for a single deck ship. These should be the same values since it's a single deck.